This works locally for me with a personal API token with scope `array:admin`. However, when the action itself runs on my fork with an API token saved as `TILEDB_CLOUD_TOKEN`, it fails when running `tiledb.array_exists(uri)`

```python
/home/runner/work/_temp/pyenv/lib/python3.10/site-packages/tiledb/cloud/config.py:96: UserWarning: You must first login before you can run commands. Please run tiledb.cloud.login.
  warnings.warn(
Info: Storage credential name is IsaiahCloudKey
Info: Notebook to upload is ./upload-notebooks/notebook-example.ipynb
Info: TileDB Cloud URI is tiledb://TileDBCITesting/s3://tiledb-test/tiledb-cloud-upload-ci-testing/notebook-example
Traceback (most recent call last):
  File "/home/runner/work/github-actions/github-actions/./upload-notebooks/upload-notebooks.py", line 118, in <module>
    if tiledb.array_exists(uri):
  File "/home/runner/work/_temp/pyenv/lib/python3.10/site-packages/tiledb/highlevel.py", line 149, in array_exists
    if tiledb.object_type(uri) != "array":
  File "tiledb/libtiledb.pyx", line 3171, in tiledb.libtiledb.object_type
  File "tiledb/libtiledb.pyx", line 374, in tiledb.libtiledb.check_error
  File "tiledb/libtiledb.pyx", line 368, in tiledb.libtiledb._raise_ctx_err
  File "tiledb/libtiledb.pyx", line 353, in tiledb.libtiledb._raise_tiledb_error
tiledb.cc.TileDBError: [TileDB::REST] Error: Error in libcurl OPTIONS operation: libcurl error message 'CURLE_OK'; HTTP code 401; server response was empty.
```

--- a/.github/workflows/upload-notebooks.yml
+++ b/.github/workflows/upload-notebooks.yml
@@ -1,0 +1,45 @@
+name: Test action upload-notebooks
+on:
+  push:
+    paths:
+      - '.github/workflows/upload-notebooks.yml'
+      - 'upload-notebooks/**'
+  pull_request:
+    paths:
+      - '.github/workflows/upload-notebooks.yml'
+      - 'upload-notebooks/**'
+  workflow_dispatch:
+jobs:
+  upload-notebooks-script:
+    runs-on: ${{ matrix.os }}
+    name: Test upload-notebooks.py (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Install tiledb-cloud
+      run: pip3 install tiledb "tiledb-cloud>=0.10.1"
+    - name: Test script
+      env:
+        TILEDB_CLOUD_TOKEN: ${{ secrets.TILEDB_CLOUD_TOKEN }}
+      run: cd upload-notebooks && bash test-upload-notebooks.sh
+  upload-notebooks-action:
+    runs-on: ${{ matrix.os }}
+    name: Test upload-notebooks action (${{ matrix.os }})
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v3
+    - name: Upload notebooks
+      uses: ./upload-notebooks
+      with:
+        notebooks_local: ./upload-notebooks/notebook-example.ipynb
+        notebooks_remote: tiledb://TileDBCITesting/s3://tiledb-test/tiledb-cloud-upload-ci-testing/notebook-example
+        TILEDB_CLOUD_TOKEN: ${{ secrets.TILEDB_CLOUD_TOKEN }}
+        TILEDB_CLOUD_STORAGE_CREDENTIAL_NAME: IsaiahCloudKey

--- a/upload-notebooks/action.yml
+++ b/upload-notebooks/action.yml
@@ -26,15 +26,15 @@ runs:
   steps:
     - name: Install tiledb-cloud package in virtual env
       run: |
-        python -m venv "$RUNNER_TEMP/pyenv"
-        "$RUNNER_TEMP/pyenv/bin/pip" install "tiledb-cloud>=0.10.1"
+        python3 -m venv "$RUNNER_TEMP/pyenv"
+        "$RUNNER_TEMP/pyenv/bin/pip" install tiledb "tiledb-cloud>=0.10.1"
         "$RUNNER_TEMP/pyenv/bin/pip" list
       shell: bash
     - name: Upload notebooks
       env:
         TILEDB_CLOUD_TOKEN: ${{ inputs.TILEDB_CLOUD_TOKEN }}
       run: |
-        "$RUNNER_TEMP/pyenv/bin/python" \
+        "$RUNNER_TEMP/pyenv/bin/python3" \
           "${{ github.action_path }}/upload-notebooks.py" \
           --notebooks-local ${{ inputs.notebooks_local }} \
           --notebooks-remote ${{ inputs.notebooks_remote }} \

--- a/upload-notebooks/test-upload-notebooks.sh
+++ b/upload-notebooks/test-upload-notebooks.sh
@@ -1,18 +1,23 @@
 #!/bin/bash
 set -u
 
+set -e
+
 echo -e "\n===\nTest: Upload minimal example notebook\n==="
 NOTEBOOK_EXAMPLE="notebook-example-$RANDOM.ipynb"
 cp notebook-example.ipynb $NOTEBOOK_EXAMPLE
-python upload-notebooks.py \
+python3 upload-notebooks.py \
   --notebooks-local $NOTEBOOK_EXAMPLE \
-  --notebooks-remote tiledb://jdblischak/s3://tiledb-cloud-jdblischak/notebooks/${NOTEBOOK_EXAMPLE%.ipynb} \
-  --tiledb-cloud-storage-credential-name personal-aws \
+  --notebooks-remote tiledb://TileDBCITesting/s3://tiledb-test/tiledb-cloud-upload-ci-testing/${NOTEBOOK_EXAMPLE%.ipynb} \
+  --tiledb-cloud-storage-credential-name IsaiahCloudKey \
   --delete-remote-notebooks
 rm $NOTEBOOK_EXAMPLE
 
+# The below examples purposefully trigger errors
+set +e
+
 echo -e "\n===\nTest: Fail for missing remote URI \n==="
-python upload-notebooks.py \
+python3 upload-notebooks.py \
   --notebooks-local \
     notebook-1.ipynb \
     notebook-2.ipynb \
@@ -20,13 +25,13 @@ python upload-notebooks.py \
     tiledb://namespace/s3://notebook-1
 
 echo -e "\n===\nTest: Fail for missing notebook\n==="
-python upload-notebooks.py \
+python3 upload-notebooks.py \
   --notebooks-local missing.ipynb \
   --notebooks-remote tiledb://namespace/s3://missing
 
 echo -e "\n===\nTest: Fail for wrong extension\n==="
 touch wrong-extension.txt
-python upload-notebooks.py \
+python3 upload-notebooks.py \
   --notebooks-local wrong-extension.txt \
   --notebooks-remote tiledb://namespace/s3://wrong-extension
 rm wrong-extension.txt

--- a/upload-notebooks/upload-notebooks.py
+++ b/upload-notebooks/upload-notebooks.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Upload notebook to TileDB Cloud
 #
@@ -94,6 +94,7 @@ if token is None:
 
 # Login
 tiledb.cloud.login(token=token)
+tiledb.default_ctx(tiledb.cloud.Config())
 if storage_credential_name is not None:
     sys.stderr.write(
         "Info: Storage credential name is %s\n" % (storage_credential_name)
@@ -114,11 +115,16 @@ for i in range(len(notebooks_local)):
         )
         sys.exit(1)
 
-    with tiledb.scope_ctx(tiledb.cloud.Ctx()):
-        if tiledb.array_exists(uri):
-            sys.stderr.write("Info: Updating existing notebook file\n")
-        else:
-            sys.stderr.write("Info: Uploading new notebook file\n")
+    # This command will fail in the following situation: 1) array:admin scoped
+    # token, 2) tiledb-py 0.21.3+, 3) non-existent notebook
+    try:
+        exists = tiledb.array_exists(uri)
+    except:
+        exists = False
+    if exists:
+        sys.stderr.write("Info: Updating existing notebook file\n")
+    else:
+        sys.stderr.write("Info: Uploading new notebook file\n")
 
     uploaded = tiledb.cloud.upload_notebook_from_file(
         ipynb_file_name=notebook,
@@ -131,6 +137,5 @@ for i in range(len(notebooks_local)):
     sys.stderr.write("Info: Uploaded to %s\n" % (uploaded))
 
     if delete_remote_notebooks:
-        with tiledb.scope_ctx(tiledb.cloud.Ctx()):
-            tiledb.Array.delete_array(uri)
+        tiledb.Array.delete_array(uri)
         sys.stderr.write("Info: Deleted remote notebook %s\n" % (uri))

